mrp_schedule/rotate.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) All Rights Reserved 2015 Akretion
+#    @author David BEAL <david.beal@akretion.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+
+from openerp.osv import orm
+
+
+def get_neighbor_element(my_itr, initial=None):
+    try:
+        val = my_itr.next()
+        if initial:
+            while val != initial:
+                val = my_itr.next()
+            val = my_itr.next()
+        return val
+    except StopIteration:
+        return None
+
+
+class AbstractSelectionRotate(orm.Model):
+    _name = 'abstract.selection.rotate'
```

Comment:
Why it doesn't iterate with the "unable" value?
It is possible to make a buton to switch in form view instead of a wizard?

For a module with this feature i suggest 'selection_iteration' what do you think?